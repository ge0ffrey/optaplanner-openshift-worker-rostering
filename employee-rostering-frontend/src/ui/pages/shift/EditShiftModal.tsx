/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { connect } from 'react-redux';

import { Button, Switch, InputGroup, Label, Form, Modal } from '@patternfly/react-core';
import DatePicker from 'react-datepicker';

import Shift from 'domain/Shift';
import Spot from 'domain/Spot';
import Employee from 'domain/Employee';
import { AppState } from 'store/types';
import { employeeSelectors } from 'store/employee';
import { spotSelectors } from 'store/spot';

import "react-datepicker/dist/react-datepicker.css";
import TypeaheadSelectInput from 'ui/components/TypeaheadSelectInput';

interface Props {
  shift?: Shift;
  isOpen: boolean;
  employeeList: Employee[];
  spotList: Spot[];
  onSave: (shift: Shift) => void;
  onClose: () => void;
}

const mapStateToProps = (state: AppState, ownProps: {
  shift?: Shift;
  isOpen: boolean;
  onSave: (shift: Shift) => void;
  onClose: () => void;
}): Props => ({
  ...ownProps,
  employeeList: employeeSelectors.getEmployeeList(state),
  spotList: spotSelectors.getSpotList(state)
}); 

interface State {
  editedValue: Partial<Shift>;
}

class EditShiftModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.onSave = this.onSave.bind(this);
    this.state = {
      editedValue: { ...this.props.shift }
    }
  }

  componentDidUpdate(prevProps: Props, prevState: State) {
    if (this.props.shift !== undefined && this.props.shift.id !== prevState.editedValue.id) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ editedValue: this.props.shift });
    }
  }

  onSave() {
    const shift = this.state.editedValue;
    if (shift.spot !== undefined && shift.startDateTime !== undefined &&
        shift.endDateTime !== undefined && shift.employee !== undefined &&
        shift.pinnedByUser !== undefined && shift.rotationEmployee !== undefined) {
      this.props.onSave(shift as Shift);
    }
  }

  render() {
    const dateFormat = "MMMM dd, hh:mm a";
    return (
      <Modal
        title="Edit Shift"
        onClose={this.props.onClose}
        isOpen={this.props.isOpen}
        actions={
          [
            <Button key={0} onClick={this.props.onClose}>Close</Button>,
            <Button key={1} onClick={this.onSave}>Save</Button>
          ]
        }
        isSmall
      >
        <Form id="modal-element" onSubmit={(e) => e.preventDefault()}>
          <InputGroup>
            <Label>Shift Start</Label>
            <DatePicker
              selected={this.state.editedValue.startDateTime}
              onChange={date => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, startDateTime: (date !== null)? date : undefined  }
              }))}
              dateFormat={dateFormat}
              showTimeSelect
            />
          </InputGroup>
          <InputGroup>
            <Label>Shift End</Label>
            <DatePicker
              selected={this.state.editedValue.endDateTime}
              onChange={date => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, endDateTime: (date !== null)? date : undefined  }
              }))}
              dateFormat={dateFormat}
              showTimeSelect
            />
          </InputGroup>
          <InputGroup>
            <Label>Spot</Label>
            <TypeaheadSelectInput
              emptyText="Select a Spot"
              defaultValue={this.props.shift? this.props.shift.spot : undefined}
              options={this.props.spotList}
              optionToStringMap={spot => spot.name}
              onChange={spot => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, spot: spot }
              }))}
            />
          </InputGroup>
          <InputGroup>
            <Label>Employee</Label>
            <TypeaheadSelectInput
              emptyText="Unassigned"
              defaultValue={this.props.shift? this.props.shift.employee : undefined}
              options={this.props.employeeList}
              optionToStringMap={employee => employee? employee.name : "Unassigned"}
              onChange={employee => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, employee: (employee !== undefined)? employee : null }
              }))}
            />
          </InputGroup>
          <InputGroup>
            <Label>Rotation Employee</Label>
            <TypeaheadSelectInput
              emptyText="None"
              defaultValue={this.props.shift? this.props.shift.rotationEmployee : undefined}
              options={this.props.employeeList}
              optionToStringMap={employee => employee? employee.name : "None"}
              onChange={employee => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, rotationEmployee: (employee !== undefined)? employee : null }
              }))}
            />
          </InputGroup>
          <InputGroup>
            <Label>Is Pinned</Label>
            <Switch
              aria-label="Is Pinned"
              isChecked={this.state.editedValue.pinnedByUser}
              onChange={isPinned => this.setState(prevState => ({
                editedValue: { ...prevState.editedValue, pinnedByUser: isPinned }
              }))}
            />
          </InputGroup>
        </Form>
      </Modal>
    );
  }
}

export default connect(mapStateToProps)(EditShiftModal);