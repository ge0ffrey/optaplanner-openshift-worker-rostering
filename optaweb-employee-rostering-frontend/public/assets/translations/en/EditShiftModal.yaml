---
createShift: Create Shift
editShift: Edit Shift
shiftStart: Shift Start
shiftEnd: Shift End
addtionalSkills: Additional Skills
selectAdditionalSkills: Select Additional Skills...
originalEmployee: Original Employee
none: None
isPinned: Is Pinned