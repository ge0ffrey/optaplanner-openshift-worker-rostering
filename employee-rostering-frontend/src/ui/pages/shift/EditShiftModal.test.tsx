/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import Spot from 'domain/Spot';
import Employee from 'domain/Employee';
import Shift from 'domain/Shift';
import { EditShiftModal } from './EditShiftModal';
import moment from 'moment';

describe('Edit Shift Modal', () => {
  it('should render correctly when closed', () => {
    const editShiftModal = shallow(<EditShiftModal
      {...baseProps}
      isOpen={false}
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftModal)).toMatchSnapshot();
  });

  it('should render correctly when opened', () => {
    const editShiftModal = shallow(<EditShiftModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftModal)).toMatchSnapshot();
  });

  it('should render correctly with a shift when opened', () => {
    const editShiftModal = shallow(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftModal)).toMatchSnapshot();
  });

  it('should update state to match shift when props change', () => {
    const editShiftModal = shallow(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);

    editShiftModal.setProps({
      shift: baseShift
    });
    expect(editShiftModal.state("editedValue")).toEqual(baseShift);

    editShiftModal.setState({
      editedValue: {
        ...baseShift,
        pinnedByUser: true
      }
    });
    expect(editShiftModal.state("editedValue")).toEqual({
      ...baseShift,
      pinnedByUser: true
    });

    editShiftModal.setProps({
      shift: undefined
    });
    expect(editShiftModal.state("editedValue")).toEqual({
      tenantId: baseProps.tenantId,
      employee: null,
      rotationEmployee: null,
      pinnedByUser: false
    });
  });

  it('should call onSave iff all required properties are defined', () => {
    const onSave = jest.fn();
    const editShiftModal = new EditShiftModal({
      ...baseProps,
      onSave: onSave,
      onClose: jest.fn(),
      onDelete: jest.fn(),
      isOpen: false
    });

    editShiftModal.state = {
      editedValue: {

      }
    };
    editShiftModal.onSave();
    expect(onSave).not.toBeCalled();

    editShiftModal.state = {
      editedValue: {
        employee: null,
        spot: spot
      }
    };
    editShiftModal.onSave();
    expect(onSave).not.toBeCalled();

    editShiftModal.state = {
      editedValue: baseShift
    };
    editShiftModal.onSave();
    expect(onSave).toBeCalled();
    expect(onSave).toBeCalledWith(baseShift);
  });

  it('should call onSave when the save button is clicked with a completed shift', () => {
    const onSave = jest.fn();
    const editShiftModal = mount(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={onSave}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);

    editShiftModal.find('button[aria-label="Save"]').simulate("click");
    expect(onSave).toBeCalled();
    expect(onSave).toBeCalledWith(baseShift);
  });

  it('should call onClose when the bottom close button is clicked', () => {
    const onClose = jest.fn();
    const editShiftModal = mount(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={onClose}
      onDelete={jest.fn()}
    />);
    
    editShiftModal.find('button[aria-label="Close Modal"]').simulate("click");
    expect(onClose).toBeCalled();
  });

  it('should call onClose when the top right X close button is clicked', () => {
    const onClose = jest.fn();
    const editShiftModal = mount(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={onClose}
      onDelete={jest.fn()}
    />);
    
    editShiftModal.find('button[aria-label="Close"]').simulate("click");
    expect(onClose).toBeCalled();
  });

  it('should call onDelete when the delete button is clicked', () => {
    const onDelete = jest.fn();
    const editShiftModal = mount(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={onDelete}
    />);
    
    editShiftModal.find('button[aria-label="Delete"]').simulate("click");
    expect(onDelete).toBeCalled();
  });

  it('should prevent changing URL when completed', () => {
    const event = {
      preventDefault: jest.fn()
    }
    const editShiftModal = shallow(<EditShiftModal
      {...baseProps}
      shift={baseShift}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftModal.find("Form").simulate("submit", event);
    expect(event.preventDefault).toBeCalled();
  });
});

const spot: Spot = {
  tenantId: 0,
  id: 2,
  version: 0,
  name: "Spot",
  requiredSkillSet: [
    {
      tenantId: 0,
      id: 3,
      version: 0,
      name: "Skill"
    }
  ]
}

const employee: Employee = {
  tenantId: 0,
  id: 4,
  version: 0,
  name: "Employee 1",
  contract: {
    tenantId: 0,
    id: 5,
    version: 0,
    name: "Basic Contract",
    maximumMinutesPerDay: 10,
    maximumMinutesPerWeek: 70,
    maximumMinutesPerMonth: 500,
    maximumMinutesPerYear: 6000
  },
  skillProficiencySet: [{
    tenantId: 0,
    id: 6,
    version: 0,
    name: "Not Required Skill"
  }]
}

const baseShift: Shift = {
  tenantId: 0,
  id: 1,
  version: 0, 
  startDateTime: moment("2018-07-01T09:00").toDate(),
  endDateTime: moment("2018-07-01T17:00").toDate(),
  spot: spot,
  employee: employee,
  rotationEmployee: {
    ...employee,
    id: 7,
    name: "Rotation Employee"
  },
  pinnedByUser: false,
  indictmentScore: { hardScore: 0, mediumScore: 0, softScore: 0 },
  requiredSkillViolationList: [],
  unavailableEmployeeViolationList: [],
  shiftEmployeeConflictList: [],
  desiredTimeslotForEmployeeRewardList: [],
  undesiredTimeslotForEmployeePenaltyList: [],
  rotationViolationPenaltyList: [],
  unassignedShiftPenaltyList: [],
  contractMinutesViolationPenaltyList: []
};

const baseProps = {
  tenantId: 0,
  spotList: [spot],
  employeeList: [employee]
}