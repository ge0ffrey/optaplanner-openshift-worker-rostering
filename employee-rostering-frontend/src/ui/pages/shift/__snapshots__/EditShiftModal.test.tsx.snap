// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Shift Modal should render correctly when closed 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Close
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Save
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={false}
  isSmall={true}
  onClose={[MockFunction]}
  title="Create Shift"
>
  <Form
    id="modal-element"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Shift Start
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift Start"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Shift End
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift End"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Spot
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        emptyText="Select a Spot"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        emptyText="Unassigned"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Rotation Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Rotation Employee"
        emptyText="None"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Is Pinned
      </Label>
      <Switch
        aria-label="Is Pinned"
        className=""
        id=""
        isChecked={true}
        isDisabled={false}
        label=""
        onChange={[Function]}
      />
    </InputGroup>
  </Form>
</Modal>
`;

exports[`Edit Shift Modal should render correctly when opened 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Close
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Save
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[MockFunction]}
  title="Create Shift"
>
  <Form
    id="modal-element"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Shift Start
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift Start"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Shift End
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift End"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Spot
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        emptyText="Select a Spot"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        emptyText="Unassigned"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Rotation Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Rotation Employee"
        emptyText="None"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Is Pinned
      </Label>
      <Switch
        aria-label="Is Pinned"
        className=""
        id=""
        isChecked={true}
        isDisabled={false}
        label=""
        onChange={[Function]}
      />
    </InputGroup>
  </Form>
</Modal>
`;

exports[`Edit Shift Modal should render correctly with a shift when opened 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Close
      </Button>,
      <Button
        aria-label="Delete"
        onClick={[Function]}
        variant="danger"
      >
        Delete
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Save
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[MockFunction]}
  title="Edit Shift"
>
  <Form
    id="modal-element"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Shift Start
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift Start"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2018-07-01T13:00:00.000Z}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Shift End
      </Label>
      <DatePicker
        allowSameDay={false}
        aria-label="Shift End"
        dateFormat="MMMM dd, hh:mm a"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2018-07-01T21:00:00.000Z}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Spot
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        defaultValue={
          Object {
            "id": 2,
            "name": "Spot",
            "requiredSkillSet": Array [
              Object {
                "id": 3,
                "name": "Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
        emptyText="Select a Spot"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        defaultValue={
          Object {
            "contract": Object {
              "id": 5,
              "maximumMinutesPerDay": 10,
              "maximumMinutesPerMonth": 500,
              "maximumMinutesPerWeek": 70,
              "maximumMinutesPerYear": 6000,
              "name": "Basic Contract",
              "tenantId": 0,
              "version": 0,
            },
            "id": 4,
            "name": "Employee 1",
            "skillProficiencySet": Array [
              Object {
                "id": 6,
                "name": "Not Required Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
        emptyText="Unassigned"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Rotation Employee
      </Label>
      <TypeaheadSelectInput
        aria-label="Rotation Employee"
        defaultValue={
          Object {
            "contract": Object {
              "id": 5,
              "maximumMinutesPerDay": 10,
              "maximumMinutesPerMonth": 500,
              "maximumMinutesPerWeek": 70,
              "maximumMinutesPerYear": 6000,
              "name": "Basic Contract",
              "tenantId": 0,
              "version": 0,
            },
            "id": 7,
            "name": "Rotation Employee",
            "skillProficiencySet": Array [
              Object {
                "id": 6,
                "name": "Not Required Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
        emptyText="None"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Is Pinned
      </Label>
      <Switch
        aria-label="Is Pinned"
        className=""
        id=""
        isChecked={false}
        isDisabled={false}
        label=""
        onChange={[Function]}
      />
    </InputGroup>
  </Form>
</Modal>
`;
